src/snapred/backend/recipe/algorithm/PixelGroupingParametersCalculationAlgorithm.py

```diff
@@ -72,15 +80,32 @@ def PyExec(self):
         self.chopIngredients(ingredients)
         self.unbagGroceries(ingredients)
 
-        # create a dummy grouped-by-detector workspace from the grouping workspace
+        # create a copy of the grouping workspace with the correct detector mask flags
+        tmpGroupingWSName = mtd.unique_hidden_name()
+        self.mantidSnapper.CloneWorkspace(
```

Comment:
[MakeDirtyDish](https://github.com/neutrons/SNAPRed/blob/578236ba553de2b2fd6f2be1529c15b50bb51922/src/snapred/backend/recipe/algorithm/MakeDirtyDish.py)?

Reply:
Unfortunately, `MakeDirtyDish` only makes a clone if `CIS_mode == True`.  I don't consider cloning an "input parameter" that will be modified to be optional.  (Obviously this can be argued both ways, but my years of frustrating experience maintaining code that did _not_ follow this practice has lead me to this assertion.)

Reply:
Oh, I guess that is right.  For some reason I was under the impression that it was just a book keeping wrapper.

Reply:
MakeDirtyDish is probably mostly obsolete at this point.  It's goal was to help Malcolm diagnose individual algorithms by logging intermediate workspaces.  We probably don't need most of them anymore.